xregexp-all.js

```diff
@@ -298,7 +300,7 @@ module.exports = function(XRegExp) {
             // Using `XRegExp.union` safely rewrites backreferences in `left` and `right`
             esc = new RegExp(
                 '(?:' + escapeChar + '[\\S\\s]|(?:(?!' +
-                    XRegExp.union([left, right]).source +
+                    XRegExp.union([left, right], basicFlags, {conjunction: 'or'}).source +
```

Comment:
Oops, I wasn't sure which flags to use, so I went with what was already used elsewhere. However, I definitely didn't consider the implications of it as thoroughly as you have, and I see now that the empty list of flags is the way to go. Thanks for merging anyway and fixing things up after the fact!